mathml/relations/html5-tree/css-inline-style-interface.tentative.html

```diff
@@ -0,0 +1,45 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8" />
+    <title>MathML 'ElementCSSInlineStyle` Mixin Tests</title>
+    <link
+      rel="help"
+      href="https://mathml-refresh.github.io/mathml-core/#dom-and-javascript"
+    />
+    <style>
+        math * { background-color: red; }
+    </style>
+    <meta name="assert" content="MathMLElements incorporate a functional ElementCSSInlineStyle interface" />
+    <script src="/mathml/support/mathml-fragments.js"></script>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+
+  </head>
+  <body>
+    <span>This tests the presence and functionality of features of the `ElementCSSInlineStyle` interface for MathMLElements</span>
+    <math>
+      <mi>E</mi>
+    </math>
+  </body>
+  <script>
+
+    let mathEl = document.querySelector("math");
+
+    Object.keys(MathMLFragments).forEach((elName) => {
+      mathEl.innerHTML = MathMLFragments[elName]
+      mathEl.style.backgroundColor = "green"
+      mathEl.style.display = "block"
```

Comment:
Why do you need do set the display?

I'm also not sure why you set backgroundColor on the math tag.

Reply:
Both are related to the fact that I didn't want to get intertwined in what fragments will be created and whether they will be valid or yield some display none.  By setting it to block explicitly it seems to me I'm testing the actual thing rather than a potential side effect - what we care about here is the API works right? As long as it is display block, we can know that - but if it was somehow marked as display none, maybe we couldn't.

As to why the background color on the math tag itself it's because then if something fails I can look at the actual rendering and see if it is red or green instead of blue and figure out what's gone astray.  Is it harmful somehow?

Reply:
I guess the idea is to try to minimize the tests so it's easier to debug.

I guess it's fine for the background color if it's only for debugging purpose. My comment was because you are setting the style on mathEl (math root) not on the element tested.

For the "display: block" you are actually introducing more complexity (e.g. that's probably not really relevant here but if in the future we introduce new display for MathML elements, you are actually changing the layout nodes used...) and that's not necessary with the fragments you are testing.

FWIW, all the fragments are designed to be valid MathML.

Removing "display: none" would only be useful for MathMLFragments == annotation or annotation-xml but you are not testing them. You should really use FragmentHelper.element() to get the proper element. Or otherwise use FragmentHelper.isValidChildOfMrow() to skip those that won't be direct child of mathEl.

Reply:
I changed this by switching the math root to have its own separate test - it's better and achieves the same thing as the debugging issue I mentioned.

But, the display block I don't think I agree... I don't care what the display is here, as long as it isn't none, this same test of the API should really work for every element in html (hidden, meta, script, style, etc are all actually displayable)... I'm kind of surprised that there isn't a thing like the globals test already.  I can change it if you like, but it feels like it is then buying more into assumptions and adding complexity.  All we care about in this test is whether the .style property exists and that we can use it to do something, I think.

Reply:
OK, background sounds better.

So just to clarify, you are doing something wrong that must be fixed here before landing:

mathEl.innerHTML = MathMLFragments[elName]

adds a fragment as a child of the math root and then you are testing 

el = mathEl.firstElementChild

which is the element corresponding to elName, only when that element can be a child of an mrow. So you should either skip elTag with !FragmentHelper.isValidChildOfMrow(elTag) or access the actual element to test using FragmentHelper.element(mathEl).

Now regarding the display, you are the one adding assumptions and complexity. Why would a MathML element be "display: none"? If you can't answer to that question, then there is no reason to modify the default display value. The reference must be the core spec, which says that only the non-first children of maction or semantics will have display: none by default:

https://mathml-refresh.github.io/mathml-core/#user-agent-stylesheet

But as explained above, you are not testing them! If at the end you choose to test them and if "display: none" is really causing problems for your tests, then you would need to add this change but only for elTag == annotation or annotation-xml. Otherwise you are no longer keeping the "minimal change" paradigm for other elements. Maybe the display is not important for your purely DOM tests here, but for things involving layout that would definitely cause issues.

In any case, you need to add a comment clearly explaining why "display: none" is needed and referring to the MathML core spec, since that's what browsers should follow. From your reply, it does not seem you actually encountered any concrete problem but just assuming potential ones.

Reply:
Alright, I think I updated this to how you suggest it should work - this isn't a hill I'm prepared to defend for no real reason.

All I was suggesting is that none of that seems necessary - if you just had a list of tag names, any tag names, mathml or not, this approach should work for this test - it doesn't even have to be valid, or even a descendant of math afaik -- all that is technically necessary, it seems, is that it is in the tree and not display none (as far as I know).  Here, it seems we need to know a lot more, how to create valid mathml trees and work with an API which I don't get to control what it creates or assumes....  But, none of that ultimately really matters, I'd just like to have tests... So, does this work for you? Did I use the fragments API correctly?